libs/block_generator/block_generator.cpp

```diff
@@ -254,4 +255,44 @@ namespace data_types::block_generator {
         fill_block(json_value, res);
         return res;
     }
+
+    sequence_generator::sequence_generator(const std::string &config_file_name) {
+        std::ifstream config_file(config_file_name.c_str());
+        if (!config_file.is_open()) {
+            std::cerr << "Could not open the config file " << config_file_name << std::endl;
+            return;
+        }
+        std::expected<boost::json::value, std::string> parsed_json =
+            json_helpers::parse_json(config_file);
+        if (!parsed_json) {
+            std::cerr << "Parsing of JSON config file failed: " << parsed_json.error() << std::endl;
+            return;
+        }
+        m_config = parsed_json.value();
+        auto validation_err = check_json_config(m_config);
+        if (validation_err) {
+            std::cerr << "Validation of JSON config file failed: " << validation_err.value()
+                      << std::endl;
+        }
+    }
+
+    int sequence_generator::generate_block(unsigned char *block_data) {
+        if (!m_config.is_object()) {
+            std::cerr << "Config is empty";
+            return -1;
+        }
+        data_types::Block res;
+        fill_block(m_config, res);
```

Comment:
Seems like this whole piece of code could be replaced with 
`res = data_types::block_generator::generate_block(m_config);` 
Do we really need this re-implementation?